### DISCUSSION TITLE
Where does `useRuntimeConfig` come from?

### DISCUSSION THREAD

**Discussion by joewagner** [nuxt/nuxt#17677] [Questions] [answered]
Hey all!  
I have a question that is related to this discussion: https://github.com/nuxt/framework/discussions/3726

**Context:**  I built an app with nuxt3 about 3 weeks ago, before the newest RC. The app uses the options api and a global store pattern via Pinia.  At the time I built it the `useRuntimeConfig` function was accessable via `import { useRuntimeConfig } from '#nitro'` with the new release candidate it doesn't seem to be possible to explicitly import it anymore. I tried importing from `'#app'` and `'#nitro'` but neither work.

**2 Questions:** why is `useRuntimeConfig` "auto-magically" injected into the `state` function but not importable?  Is there a different, better, way to get the runtime config Object?

The full app is located here: https://github.com/tablelandnetwork/example-apps/tree/main/gila/frontend
It's still a work in progress, and any suggestions are very welcome!

Thanks to anyone who might be able to answer my questions,

**Comment by pi0** [comment, marked as answer]
Auto imports should normally work but have you tried `import { useRuntimeConfig } from '#imports'`?

**Reply by joewagner** [reply]
`import { useRuntimeConfig } from '#imports'` works. Thanks!  However a different issue arises it you try to call `useRuntimeConfig` immediately after the import.  
```
nuxt.mjs:119 Uncaught Error: nuxt instance unavailable
```
afaict this is because of how I'm setting up the pinia store.  Regardless, putting the `useRuntimeConfig` call inside the state function works.  Thanks again for the help
